inbm-lib/inbm_common_lib/utility.py

```diff
@@ -138,7 +138,7 @@ def create_file_with_contents(path: Union[str, Path], contents: List[str]) -> No
         raise IOError(f"Error while writing file: {str(e)}")
```

Comment:
**Suggestion:** Instead of converting exceptions to strings and then wrapping them in IOError, directly raise the original exception to preserve the stack trace. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        except (PermissionError, IsADirectoryError, OSError) as e:
            raise IOError("Error while writing file") from e
```